src/SonarScanner.MSBuild.PreProcessor/WebClientDownloader.cs

```diff
@@ -88,34 +88,27 @@ public async Task<bool> IsLicenseValid(string url)
 
             var content = await response.Content.ReadAsStringAsync();
 
-            if (response.StatusCode == HttpStatusCode.NotFound)
-            {
-                var json = JObject.Parse(content);
 
-                var jsonErrors = json["errors"];
-
-                if (jsonErrors?.Any(x => x["msg"]?.Value<string>() == "License not found") == true)
-                {
-                    return false;
-                }
+            if (response.StatusCode == HttpStatusCode.NotFound || response.StatusCode == HttpStatusCode.Unauthorized)
+            {
+                // In the case of the Community Edition, the request may fail with one of the following status code:
+                //  - HttpStatusCode.NotFound - if "Force user authentication" is disabled
+                //  - HttpStatusCode.Unauthorized - if "Force user authentication" is enabled
+                //
+                // In the case of other editions, the request should not be authenticated and should exists so it will not met the above condition
```

Comment:
This is not true, commercial editions also return 404 on some cases.

Reply:
Ok, so should I return back the check for "License not found" error messages or should you prefer to check the edititon by calling api/system/info?

Reply:
I've returned back the msg check